### ISSUE TITLE
Import upsert can drop issue child history; write probe accepts zero-row update

### ISSUE THREAD

**Issue by larusivar** [Dicklesworthstone/beads_rust#263]
## Summary

I found two storage correctness issues on current `main` while testing the JSONL import / write-probe paths:

1. `upsert_issue_for_import()` updates an existing issue by doing `DELETE FROM issues WHERE id = ?` followed by `INSERT`. Because child tables reference `issues(id)` with cascading foreign keys, an import update can drop existing issue history such as events, labels, dependencies, and comments that should remain attached to the issue.
2. `finish_issue_mutation_write_probe()` treats `Ok(0)` from the probe update as success. A zero-row probe means the issue was not actually write-addressable through that mutation path, so accepting it hides the failure the probe is meant to catch.

## Why this matters

The import issue can silently lose relational history during import/rebuild of an existing issue: the parent issue comes back, but child rows that recorded or related to the previous parent row are removed by the delete before the replacement insert.

The probe issue can make doctor/probe-style diagnostics report a write path as healthy even when the update matched no row. That is especially risky around database visibility/write-addressability bugs because it turns a diagnostic into a false negative.

## Repro shape

The import issue is reproducible with a simple in-memory storage test:

1. Create an issue normally, which creates at least one event row referencing the issue.
2. Clone that issue, change the title/updated_at, and call `upsert_issue_for_import()`.
3. Confirm the issue title changed.
4. Confirm the pre-existing event rows are still present and `has_missing_issue_reference("events", "issue_id")` is false.

With the current delete+insert behavior, the child-row preservation assertion fails because the existing event row is removed before the parent is reinserted.

The probe issue is smaller:

```rust
finish_issue_mutation_write_probe(Ok(0), Ok(0))
```

Currently this returns success; I would expect it to return an error because the probe did not find/update the target issue.

## Expected behavior

- Import upsert should update existing `issues` rows in place and only insert when the issue does not exist.
- The existence check for import upsert should not parse the existing row before overwriting it; import should be able to heal a malformed current row from valid JSONL.
- A mutation write probe should treat zero matched rows as failure, even if rollback succeeds.
- If both the probe update matches zero rows and rollback fails, the zero-row probe result should remain the primary diagnostic because it is the reason the write path is not trustworthy.

## Reference patch

I opened a minimal one-file draft/reference PR with regression tests: #264. It does not change CLI behavior or add local/fork-specific machinery; it only changes the storage helper and the write-probe result handling.

Validation on that patch:

- `cargo fmt --check`
- `ubs --files src/storage/sqlite.rs --only=rust --format=json --ci` — 0 critical findings
- `CARGO_BUILD_JOBS=2 cargo check --all-targets`
- `CARGO_BUILD_JOBS=2 cargo clippy --all-targets -- -D warnings`
- `CARGO_BUILD_JOBS=2 cargo test --lib` (`1418 passed; 0 failed`)
- `CARGO_BUILD_JOBS=2 cargo test --release e2e_sync` (`39 passed; 0 failed` for the sync-filtered tests)

**Comment by larusivar** [comment]
Linked draft/reference PR: #264. I kept the PR intentionally small and framed it as an implementation reference; the issue remains the primary report.

**Comment by Dicklesworthstone** [comment]
Confirming this is resolved on main.

Fix landed in `ab2f74d5`: `upsert_issue_for_import` now uses a narrow `SELECT 1` existence probe + in-place `UPDATE` (falling back to `INSERT`) so cascaded foreign keys no longer sweep events / labels / deps / comments out from under the parent row on import. `finish_issue_mutation_write_probe` treats `Ok(0)` as the primary failure and demotes any coexisting rollback error to a `tracing::warn!`.

Regression coverage:
- `test_upsert_issue_for_import_preserves_child_rows`
- `test_upsert_issue_for_import_overwrites_malformed_existing_row`
- `test_finish_issue_mutation_write_probe_rejects_zero_row_update`
- `test_finish_issue_mutation_write_probe_prefers_zero_row_over_rollback_error`

Thanks for the careful write-up — separating the two concerns made both clean to fix.
